### PR TITLE
Issue 973: Add initial Django Admin A11y Fixes

### LINKED ISSUE #973: Django Admin a11y Fixes (Sprint 1)
**Description:**
Delivers useful existing resources (USWDS theme for Django Admin) and optimizes them for use in our project, sets ground for testing. 

**Acceptance Criteria:** 
- [x] Django admin is more visually accessible (e.g. color, contrast) 
- [x] Testing Checklist has been run and all tests pass

**Tasks:**
- [x] Create fork of [https://github.com/fabiocaccamo/django-admin-interface](https://github.com/fabiocaccamo/django-admin-interface)
- - [x] [https://github.com/raft-tech/django-admin-interface](https://github.com/raft-tech/django-admin-interface)
- [x] Remove non-USWDS data and themes from fork
- [x] Publish initial version of fork as package to pypi under a Raft org
- [x] Install package to TANF, add to settings.py, and open PR

**Supporting Documentation:**
[https://hackmd.io/@mreiter/ryuHXhbuu](https://hackmd.io/@mreiter/ryuHXhbuu
) Contains task list for a11y fixes that will follow on from this ticket and complete #892

### PR DESCRIPTION
**Pull request by jorgegonzalez**
### Summary of Changes

This PR adds [`django-admin-508`](https://github.com/raft-tech/django-admin-508) as a dependency and replacement	for the default user interface provided by Django-Admin, in order to achieve an accessible admin UI to the 508 Standard.

_Addresses issue #973_
_Acceptance criteria as stated in the issue_

### How to Test
_List the steps to test the PR_

- Clone this branch, and build the backend application.
- Login to Django Admin in `localhost:8080/admin` and notice the changes.

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
	  - [x] Django admin is more visually accessible (e.g. color, contrast) 
	  - [x] Testing Checklist has been run and all tests pass

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

Does not apply.

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

Does not apply.

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

We should hold off full government A11y testing until #1054 is completed.

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

Does not apply.

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?
	- No issues found.

### CODE REVIEW

File: tdrs-backend/tdpservice/settings/common.py [hunk 1]
```diff
@@ -67,6 +67,8 @@ class Common(Configuration):
     """Define configuration class."""
 
     INSTALLED_APPS = (
+        "colorfield",
+        "admin_interface",
```
Comment (jorgegonzalez) [inline tdrs-backend/tdpservice/settings/common.py:71]: The mere addition of our `django-admin-508` package will supersede the default Django Admin templates.

**Comment by jorgegonzalez** [comment]
The new Admin site is currently. deployed: https://tdp-backend-sandbox.app.cloud.gov/admin/

Screen Shot 2021-07-06 at 10 57 00 AM

Note the new `"Themes"` table that this package adds. This is a leftover from the original [Django Admin Interface](https://github.com/fabiocaccamo/django-admin-interface) and I plan to hide it in the next PR following this one #1053

**Review by reitermb — APPROVED** [review]
Raft A11y Review: 

Looking great! 🚀 We'll have some follow-on contrast fixes (as expected) in #1053, but this made a big dent in issue quantity, particularly on pages with a lot of repetitive elements that were failing before like Log Entries (as seen below). 

image
image

**Review by jtwillis92 — APPROVED** [review]
LGTM, pulled it down locally and had no issues. This is a huge improvement!

Re: removing Themes - several of the features included with themes are pretty useful and I don't think it would hurt to leave it around.

In particular, the ability to designate the environment:

environment-djadmin-themes

The ability to set the logo and title easily:
Screenshot from 2021-07-09 17-20-55
Screenshot from 2021-07-09 17-21-14

**Comment by ADPennington** [comment]
looks great @jorgegonzalez!

- color much easier on the eyes ✅ 
- appreciate the movement of the toggle button to minimize the sidebar (more obvious now ➕ )
- also like the admin interface themes ( bonus ➕ )

To confirm, are any of the tasks in the [hackmd](https://hackmd.io/@mreiter/ryuHXhbuu) fully addressed w/ this PR or are those to be tackled in the follow-ons (#1053, #1054) ?

p1

**Comment by jorgegonzalez** [comment]
@ADPennington Correct, no tasks in the HackMD are fully addressed in this PR, and are being addressed in #1053, and #1054.

File: tdrs-backend/Pipfile [hunk 2]
```diff
@@ -44,6 +44,7 @@ requests = "~=2.25.0"
 django-admin-logs = "~=1.0.1"
 drf-yasg = "~=1.20.0"
 wait-for-it = "==2.2.0"
+django-admin-508 = "==0.0.2"
```
Comment (ADPennington) [inline tdrs-backend/Pipfile:47]: very cool to see [this](https://pypi.org/project/django-admin-508/0.0.2/)@jorgegonzalez!

**Review by ADPennington — APPROVED** [review]
great start to impt a11y work @jorgegonzalez! @abottoms-coder this is ready to merge 🚀 

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
  - [x] Django admin is more visually accessible (e.g. color, contrast)
  - [x] Testing Checklist has been run and all tests pass

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

_n/a: this PR introduces a branch dependency of django-admin to be customized for 508 compliance purposes_

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are backend code style checks passing on CircleCI?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*
+ [ ] Does this PR complete the epic?
_this is the first of 3 PRs to be submitted as part of epic #892_
+ [ ] Are links included to any other gov-approved PRs associated with epic?
_n/a_
+ [x] Does PR include documentation for Raft's a11y review?
https://github.com/raft-tech/TANF-app/pull/1063#pullrequestreview-701150498


### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [x] If this PR introduces backend code, is that code documented both inline and overall?
+ [x] If this PR introduces dependencies, are their licenses documented?
@jorgegonzalez 's  django-admin-508 dependency! https://github.com/raft-tech/django-admin-508/blob/main/LICENSE.txt

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?
_no new issues detected_

**Pull request merged**